### ISSUE TITLE
Typing support for functools.update_wrapper

### ISSUE THREAD

**Issue by twoertwein** [microsoft/pyright#3812]
I fully understand that the following might create a "hell no!" reaction :)

**Is your feature request related to a problem? Please describe.**
`functools.update_wrapper` can be abused to copy methods from one class to another class. This is probably really bad practice. Pandas does something similarly bad but with its own [decorator](https://github.com/pandas-dev/pandas/blob/62a69beddbedde349891378992c902c0b9341a9f/pandas/core/indexes/extension.py#L110). I assume the only way to inform type checkers that a wrapped class has methods from another class is by manually enumerating all signatures using `copied_name: Callable[...]` or `copied_name: SomeProtocol` in the body of the wrapped class.

With `LiteralString`s I could assume that it might be possible to type `functools.update_wrapper` when all of its parameters are literals, for example in this case:

```py
from functools import update_wrapper

class A:
    def cool_function_A(self) -> str:
        return "cool_function_A"

    def function_B(self) -> str:
        return "function_B"

class _B:
    def function_C(self) -> str:
        return "function_C"

B = update_wrapper(_B, A, assigned=("cool_function_A",), updated=())

print(B().cool_function_A())
print(B().function_C())

try:
    print(B().function_B())
    assert False
except AttributeError:
    pass
```

If the above would work with static type checkers, pandas could maybe use `update_wrapper` to avoid copying all the signatures.

**Describe the solution you'd like**
Typing support for `functools.update_wrapper`.

**Comment by erictraut** [comment]
Oh my.

Yeah, that's unlikely to happen.

I give you credit for exhibiting a modicum of shame for suggesting this. :)
